### PR TITLE
When pushing to a GitHub repository, use HTTPS

### PR DESCRIPTION
**Pull request by Keno**
Since we can use the token we already have for API.
Also clean a couple deprecations.

### CODE REVIEW

File: src/PkgDev.jl [hunk 1]
```diff
@@ -21,7 +21,7 @@ const cd = Pkg.Dir.cd
 Return package `pkg` directory location through search. Additional `paths` are appended.
 """
 function dir(pkg::AbstractString)
-    pkgsrc = Base.find_in_path(pkg, Pkg.dir())
+    pkgsrc = Base.find_in_path(String(pkg), Pkg.dir())
```
Comment (tkelman) [inline src/PkgDev.jl]: is this needed? shouldn't `find_in_path` be smart enough to handle this?

Comment (Keno) [inline src/PkgDev.jl]: You're right, but I wouldn't have put it if there wasn't some deprecation warning that told me so. Though I was on an old version of PkgDev, so maybe this was already fixed and rebase put it back.

Comment (Keno) [inline src/PkgDev.jl]: Ok, I suspect what happened is that I didn't have ca9fdc8583f931b06d5c2ff7eded46c2b258bdfa

**Comment by Keno** [comment]
I'm gonna go ahead and merge this. We should probably also tag a new version.

**Comment by tkelman** [comment]
Agreed. The tests from #58 might fail on PackageEvaluator, I'm not sure.

**Comment by Keno** [comment]
I think that's a fine trade-off. If it does, we'll fix it. Since other packages don't depend on this, we don't risk cascading failures.

**Comment by tkelman** [comment]
Did fail on PackageEvaluator: http://pkg.julialang.org/detail/PkgDev.html

**Pull request merged**